src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -386,10 +387,44 @@ private static IPHostEntry GetHostEntryCore(string hostName, AddressFamily addre
         private static IPAddress[] GetHostAddressesCore(string hostName, AddressFamily addressFamily, NameResolutionActivity? activityOrDefault = default) =>
             (IPAddress[])GetHostEntryOrAddressesCore(hostName, justAddresses: true, addressFamily, activityOrDefault);
 
+        private static bool ValidateAddressFamily(ref AddressFamily addressFamily, string hostName, bool justAddresses, [NotNullWhen(false)] out object? resultOnFailure)
+        {
+            if (!SocketProtocolSupportPal.OSSupportsIPv6)
```

Comment:
should this be symmetric e.g. should we apply same logic to IPv4?

Reply:
Is `OSSupportsIPv6 == false` a practical real-life user scenario? I would be hesitant to handle it otherwise. It would be virtually untestable; we don't have a `System.Net.DisableIPv4` switch, and I don't think it would make much sense to add one.